python/ray/train/v2/api/result.py

```diff
@@ -1,31 +1,71 @@
 import logging
 import os
-from dataclasses import dataclass, field
-from typing import Any, Dict, Optional, Union
+from dataclasses import dataclass
+from typing import TYPE_CHECKING, Any, Dict, Optional, Union
 
 import pandas as pd
 import pyarrow
 
 from ray.air.result import Result as ResultV1
 from ray.train.v2._internal.exceptions import TrainingFailedError
+from ray.util.annotations import Deprecated
+
+if TYPE_CHECKING:
+    from ray.train.v2._internal.execution.checkpoint.checkpoint_manager import (
+        CheckpointManager,
+    )
 
 logger = logging.getLogger(__name__)
 
 
 @dataclass
 class Result(ResultV1):
     error: Optional[TrainingFailedError]
-    # The metrics dataframe will not be supported in the new Result class.
-    metrics_dataframe: Optional["pd.DataFrame"] = field(init=False, default=None)
 
     @classmethod
     def from_path(
         cls,
         path: Union[str, os.PathLike],
         storage_filesystem: Optional[pyarrow.fs.FileSystem] = None,
     ) -> "Result":
-        raise NotImplementedError
+        raise NotImplementedError("`Result.from_path` is not implemented yet.")
 
+    @Deprecated
     @property
     def config(self) -> Optional[Dict[str, Any]]:
-        raise NotImplementedError
+        raise DeprecationWarning(
+            "The `config` property for a `ray.train.Result` is deprecated, "
+            "since it is only relevant in the context of Ray Tune."
+        )
+
+
+def _build_result(
+    checkpoint_manager: "CheckpointManager", error: Optional[TrainingFailedError]
+) -> Result:
```

Comment:
nit: This function could probably still exist in `controller.py`, which would allow us to keep all conceptual glue logic between `CheckpointManager` and `Result` in that file (which already has a lot of the connective logic).